### ISSUE TITLE
Bug: Moving images horizontally is much faster than moving them across lines

### ISSUE THREAD

**Issue by benlubas** [3rd/image.nvim#26]
https://github.com/3rd/image.nvim/assets/56943754/c3bdbbfa-89df-441e-bd0c-51f2bef69b31

see the gif for what I mean. I haven't looked at the code, but I would guess this has something to do with the way the markdown integration keeps track of images.

It seems like when moving an image to a new line, it's considered a new image and the data has to be sent to the terminal again (which takes a significant amount of time for larger image).

This issue is just to document the problem. I'd imagine it would also affect the neorg integration

**Comment by 3rd** [comment]
@benlubas Thanks again for yet another great issue, Ben!
I added caching for conversion, resize, and crop operations in https://github.com/3rd/image.nvim/commit/961e5a68998dd76bf5e25ae2d96fcf3bb1ee22ae


https://github.com/3rd/image.nvim/assets/59587503/bf7f2f35-4eb7-4bde-a025-6fd23e23f1b5


It has greatly improved the performance, previously we were doing these (and writing the same content to the same file multiple times)  on every render, which is expensive.

I also tried preventing the transmissions and keeping track of what's been transmitted and cleared by the Kitty backend, but that didn't add much as Kitty tracks internally what's been transmitted already, so I left that out to avoid complicating things.
